### ISSUE TITLE
cookbook not working with chef 12.15.16

### ISSUE THREAD

**Issue by MariiAnet** [vkhatri/chef-filebeat#135]
Hi! How can I solve this problem?

Relevant File Content:
----------------------
/var/chef/cache/cookbooks/filebeat/recipes/install_package.rb:

 20:  version_string = %w[fedora rhel amazon].include?(node['platform_family']) ? "#{node['filebeat']['version']}-#{node['filebeat']['release']}" : node['filebeat']['version']
 21:
 22:  case node['platform_family']
 23:  when 'debian'
 24:    include_recipe 'elastic_beats_repo::apt' if node['filebeat']['setup_repo']
 25:
 26:    unless node['filebeat']['ignore_version'] # ~FC023
 27>>     apt_preference 'filebeat' do
 28:        pin          "version #{node['filebeat']['version']}"
 29:        pin_priority '700'
 30:      end
 31:    end
 32:  when 'fedora', 'rhel', 'amazon'
 33:    include_recipe 'elastic_beats_repo::yum' if node['filebeat']['setup_repo']
 34:
 35:    unless node['filebeat']['ignore_version'] # ~FC023
 36:      yum_version_lock 'filebeat' do

[2018-04-18T13:11:16+02:00] ERROR: No resource or method named `apt_preference' for `Chef::Recipe "install_package"'

**Comment by vkhatri** [comment]
The error is caused by `apt` cookbook dependency in cookbook `elastic_beats_repo`.

```shell
diff --git a/metadata.rb b/metadata.rb
index a1c4999..6c069b7 100644
--- a/metadata.rb
+++ b/metadata.rb
@@ -9,7 +9,7 @@ source_url 'https://github.com/vkhatri/chef-elastic-beats-repo' if respond_to?(:
 issues_url 'https://github.com/vkhatri/chef-elastic-beats-repo/issues' if respond_to?(:issues_url)
 chef_version '>= 12.1' if respond_to?(:chef_version)

-depends 'apt'
+depends 'apt', '~> 6.1.4'
 depends 'yum', '>= 5.0.1'
```

I will be releasing new version of beats cookbooks later today.

**Comment by vkhatri** [comment]
Fixed in release v1.5.0.

**Comment by MariiAnet** [comment]
Thank you!
